### ISSUE TITLE
`eslint/no-await-in-loop` - should not report an error when loop contains a `break` statement

### ISSUE THREAD

**Issue by DetachHead** [oxc-project/oxc#24307]
```ts
for (let retryCount = 0; retryCount < 5; retryCount++) {
  await foo(); // error: no-await-in-loop
  if (await condition()) { // error: no-await-in-loop
    break;
  }
}
```
if a loop contains a `break`, chances are it's intentional to avoid doing unnecessary work

**Comment by camc314** [comment]
hi! Thanks for reporting this issue.

Out behaviour here aligns with eslint, we have no plans to change this behaviour.

Eslint behaviour:

https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgbm8tYXdhaXQtaW4tbG9vcDogXCJlcnJvclwiKi9cblxuZm9yIChsZXQgcmV0cnlDb3VudCA9IDA7IHJldHJ5Q291bnQgPCA1OyByZXRyeUNvdW50KyspIHtcbiAgYXdhaXQgZm9vKCk7IC8vIGVycm9yOiBuby1hd2FpdC1pbi1sb29wXG4gIGlmIChhd2FpdCBjb25kaXRpb24oKSkgeyAvLyBlcnJvcjogbm8tYXdhaXQtaW4tbG9vcFxuICAgIGJyZWFrO1xuICB9XG59Iiwib3B0aW9ucyI6eyJydWxlcyI6e30sImxhbmd1YWdlT3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e319fX19

Thanks!

**Comment by DetachHead** [comment]
could it be added as an option? there are a lot of issues with some of these old eslint rules so IMO it's a shame if oxlint doesn't want to deviate from eslint at all.

for comparison, [ruff](https://docs.astral.sh/ruff/) re-implements rules from several old python linters, but they deviate when it makes sense to do so.

**Comment by DetachHead** [comment]
another example is `while` loops:

```ts
while (true) {
  await foo(); // error: no-await-in-loop
  if (await condition()) { // error: no-await-in-loop
    break;
  }
}
```

these seem like pretty obvious false positives, i don't think oxlint needs to always mirror eslint's behavior exactly, especially when that behavior is clearly incorrect here.

alternatively, if backwards compatibility with eslint is preferred, perhaps oxlint could introduce its own version of the rule that works correctly?
